src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStackFrame.cs

```diff
@@ -112,6 +124,7 @@ public JsonConverter InitializeReEntry(Type type, JsonSerializerOptions options)
         public void Reset()
         {
             CollectionEnumerator = null;
+            AsyncEnumerator = null;
```

Comment:
I'm assuming you mean a scenario like this one?

https://github.com/dotnet/runtime/pull/50778/files#diff-8d22bc3ac93a8abd92bbd76909dbe67d81a0ba7f75a0e93f42c5b6be8e2fe83aR111

Reply:
That does has 2 properties, but do they cause a fallback to the Stream to read more data? The sceanario would be:
- Property1 runs out of data and returns back to Stream, setting the enumerator on the frame
- Property2 is enumerated and sees the enumerator from Property1 on the frame (since it wasn't nulled out)
- Property2 doesn't write anything (or perhaps throws)

When adding state that is specific to a property (like an enumerator), it is cleared in `EndProperty` not `Reset`. However, it looks like the other enumerator is also cleared in `Reset` -- perhaps that is an issue as well.

Reply:
I don't think that this scenario is possible, since `WriteStack.Push()` will make a `Reset()` call to the `Current` frame. I believe this guarantees both `CollectionEnumerator` and `AsyncEnumerator` are null on the first run of a resumable converter.

The particular test does provide scenaria where fallback to the async method is forced, and the async enumerators themselves are instrumented for creation and disposal. If it were the case that enumerators leaked to different nodes in the object graph that would manifest as either invalid serializations or assertion failures in the instrumentation.

Reply:
I see; yes we do Push() for each collection, so we are OK -- in v3.0 we didn't push for first-level collections (as an optimization), but that was changed in v5.